lib/matplotlib/backends/backend_wx.py

```diff
@@ -629,16 +630,13 @@ def gui_repaint(self, drawDC=None, origin='WX'):
             if not drawDC:
                 # not called from OnPaint use a ClientDC
                 drawDC = wx.ClientDC(self)
-
-            # following is for 'WX' backend on Windows
-            # the bitmap can not be in use by another DC,
-            # see GraphicsContextWx._cache
-            if wx.Platform == '__WXMSW__' and origin == 'WX':
-                img = self.bitmap.ConvertToImage()
-                bmp = img.ConvertToBitmap()
-                drawDC.DrawBitmap(bmp, 0, 0)
-            else:
-                drawDC.DrawBitmap(self.bitmap, 0, 0)
+            # For the wx renderer on Windows, the bitmap cannot be in use by
+            # another DC (see GraphicsContextWx._cache).
+            bmp = (self.bitmap.ConvertToImage().ConvertToBitmap()
+                   if (wx.Platform == '__WXMSW__'
+                       and isinstance(self.figure._cachedRenderer, RendererWx))
+                   else self.bitmap)
+            drawDC.DrawBitmap(bmp, 0, 0)
```

Comment:
You already did this in #18599.

Reply:
not the `origin` handling?  The two PRs will conflict, but I can rebase them in either direction after one gets merged.

Reply:
rebased